Epam/Java Basics_UpSkill Lab #1/Strings_and_basics_of_text_processing/Working_with_a_string_as_a_String_or_StringBuilder/src/com/androsor/string_object/HomeTask6.java

```diff
@@ -0,0 +1,24 @@
+package com.androsor.string_object;
+
+/**
+ * Get a new one from the given string by repeating each character twice.
+ */
+
+public class HomeTask6 {
+
+    public static void main (String [] args){
+
+        String str1;
+        str1 = HomeTask1.enterFromConsole("Введите строку");
+        StringBuilder str2 = new StringBuilder (str1.length()*2);
+
+        for (int i = 0; i<str1.length(); i++){
+
+            str2.append(str1.charAt(i));
+            str2.append(str1.charAt(i));
+
+        }
+
+        System.out.printf("Строка с задвоенными символами:\n%s", str2);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.